From f99eb7122450b426883d860d04624aeba5cc3df8 Mon Sep 17 00:00:00 2001
From: Hains van den Bosch <hainsvdbosch@ziggo.nl>
Date: Sat, 8 Sep 2018 16:33:29 +0200
Subject: [PATCH] Do not use -o option for ip busybox

Introduced with busybox 1.29.2.
With the '-o' option ip doesn't show the UP status of the networkinterface.
And thus all network interfaces are turned off.

With Busybox 1.27.2
root@dm8000:~# ip -o addr show dev eth0
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000\    link/ether 00:09:34:1c:3f:a8 brd ff:ff:ff:ff:ff:ff
2: eth0    inet 192.168.178.185/24 brd 192.168.178.255 scope global eth0
2: eth0    inet6 fe80::209:34ff:fe1c:3fa8/64 scope link \       valid_lft forever preferred_lft forever
root@dm8000:~#

With busybox 1.29.2 (with -o)
root@dm8000:~# ip -o addr show dev eth0
2: eth0    inet 192.168.178.185/24 brd 192.168.178.255 scope global eth0
2: eth0    inet6 fe80::209:34ff:fe1c:3fa8/64 scope link \       valid_lft forever preferred_lft forever

With Busybox 1.29.2 (without -o)
root@dm8000:~# ip addr show dev eth0
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether 00:09:34:1c:3f:a8 brd ff:ff:ff:ff:ff:ff
    inet 192.168.178.185/24 brd 192.168.178.255 scope global eth0
    inet6 fe80::209:34ff:fe1c:3fa8/64 scope link
    valid_lft forever preferred_lft forever
---
 lib/python/Components/Network.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/python/Components/Network.py b/lib/python/Components/Network.py
index 34b8d7bf2..56c145828 100644
--- a/lib/python/Components/Network.py
+++ b/lib/python/Components/Network.py
@@ -61,7 +61,7 @@ class Network:
 		return [ int(n) for n in ip.split('.') ]
 
 	def getAddrInet(self, iface, callback):
-		cmd = ("/sbin/ip", "/sbin/ip", "-o", "addr", "show", "dev", iface)
+		cmd = ("/sbin/ip", "/sbin/ip", "addr", "show", "dev", iface)
 		self.console.ePopen(cmd, self.IPaddrFinished, [iface, callback])
 
 	def IPaddrFinished(self, result, retval, extra_args):
-- 
2.17.1

